### PR TITLE
Add support for D1

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
## Description

Adds API support for D1 Databases
## Has your change been tested?

Added new unit tests and they pass

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

### CODE REVIEW

**Comment by Cyb3r-Jak3** [comment]
Needed for https://github.com/cloudflare/terraform-provider-cloudflare/issues/2544

File: d1.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+			break
+		}
+	}
+	return databases, &r.ResultInfo, nil
+}
+
+// CreateD1Database creates a new database for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-create-database
+func (api *API) CreateD1Database(ctx context.Context, rc *ResourceContainer, params CreateD1Parameters) (D1Database, error) {
```
Comment (jacobbednarz) [inline d1.go]: the third param should be `<Method>Params`

```suggestion
func (api *API) CreateD1Database(ctx context.Context, rc *ResourceContainer, params CreateD1DatabaseParams) (D1Database, error) {
```

File: d1.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		return D1Database{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return r.Result, nil
+}
+
+// QueryD1Database queries a database for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-query-database
+func (api *API) QueryD1Database(ctx context.Context, rc *ResourceContainer, params QueryD1Parameters) ([]D1Result, error) {
```
Comment (jacobbednarz) [inline d1.go]: the third parameter should be `<Method>Params`

```suggestion
func (api *API) QueryD1Database(ctx context.Context, rc *ResourceContainer, params QueryD1DatabaseParams) ([]D1Result, error) {
```

File: d1.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+type QueryD1Parameters struct {
+	DatabaseID string   `json:"database_id"`
+	SQL        string   `json:"sql"`
+	Parameters []string `json:"params"`
+}
+
+type D1DatabaseMetadata struct {
+	ChangedDB   bool    `json:"changed_db"`
```
Comment (jacobbednarz) [inline d1.go]: https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#booleans

```suggestion
	ChangedDB   *bool    `json:"changed_db"`
```

File: d1.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+}
+type QueryD1Response struct {
+	Result []D1Result `json:"result"`
+	Response
+}
+
+// ListD1Databases returns all databases for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-list-databases
+func (api *API) ListD1Databases(ctx context.Context, rc *ResourceContainer, params ListD1Parameters) ([]D1Database, *ResultInfo, error) {
```
Comment (jacobbednarz) [inline d1.go]: the third param should be named to `<Method>Params`

```suggestion
func (api *API) ListD1Databases(ctx context.Context, rc *ResourceContainer, params ListD1DatabasesParams) ([]D1Database, *ResultInfo, error) {
```

**Comment by jacobbednarz** [comment]
fyi @geelen

**Pull request merged**